src/box/sql/build.c

```diff
@@ -2117,6 +2169,576 @@ sql_drop_table(struct Parse *parse_context)
 	sqlSrcListDelete(db, table_name_list);
 }
 
+/** Check is BOX constraint with given name exists. */
+static bool
+is_constraint_exists(struct tuple_constraint_def *cdef, uint32_t count,
+		     const char *name)
+{
+	for (uint32_t i = 0; i < count; ++i) {
+		if (strcmp(cdef[i].name, name) == 0)
+			return true;
+	}
+	return false;
+}
+
+/** Append a new BOX constraint to array of BOX constraints. */
```

Comment:
You say it is a box constraint like it is generic, but actually use `fk` parameter name. That probably works indeed even for `CONSTR_FUNC` in theory, but I would suggest add an assertion that `fk` is `CONSTR_FKEY`. And use function name `tuple_constraint_append` or something likewise more specific. Because there are also table constraints. Also `fk` can be const.

Reply:
True, thank you. Fixed in another PR.